### ISSUE TITLE
[Brainstorming] Improving the Battle sections

### ISSUE THREAD

**Issue by macce2** [Pyrdacor/Ambermoon#102]
Hi to all Ambermoon players there!
I decided to start this new thread,
regarding the battle sequences of Ambermoon
& Ambermoon Advanced in general;
because I have a couple of requests,  
suggestions & ideas improving those.

First, I'd like to focus on the messages appearing
while on battle field:

While fighting some Curse wasps on the Forest moon,
you may get the following messages:
"curse wasp has deflected the spell (X)" and
"curse wasp is Immune to this type of spell"
successively.

If we assume that "curse wasp is immune to spell X" is true;
then the other sentence "curse wasp has deflected the spell (X)"
is totally useless and additionally kind of illogical.

So, why don't we simply remove those extraordinary sentences 
and thus maybe even free up some memory for other things ?!

The second "issue" is a small request:

while using the inventory screens during battles;
it would be nice if the character that had last been
investigated / used on the inventory screens,
would be activated automatically while returning
to the battle sequence screen.

Maybe I or someone else would like to add something to this thread
in the future, so please don't close this one yet, please !

**Comment by Pyrdacor** [comment]
I agree that not both Messages should appear together. But you still need both as immunity is based on a flag on the monster (or the element) and the deflection is a dice roll against the Anti-Magic attribute.

However if the monster is immune, the dice roll is not necessary and should not be performed.


The active player selection in battle might be useful. But I am not sure if someone who knows the original might get confused.

**Comment by macce2** [comment]
While fighting some curse wasps near Kire's dwelling in Don Kiredon;
I  was astonished of the Dungeon background appearing on the 
battle screen.
When confronted by the other insects in Don Kiredon,
the background was the city view, just like it should be there.
I think this must be fixed.

Additionally, IMHO there's some places in Ambermoon where
the background graphics of the battle screen should be changed;
or ideally, totally re-made.
Some examples:

The bandit's house (the cellar part might be ok)

ManyEyes' Castle, the roof level

etc.

Maybe this request should be moved to Ambermoon Advanced repo..!?
I do know that there propably isn't too many, if any graphicians
available for this Ambermoon project(-s) right now;
but I thought I Must write this because "the issue" has been
bothering me a considerable time;
and I even decided to start doing some sketches for some
new battle backgrounds, but then i had plenty of other things
to do and also thought that maybe im not good enough an artist to do those.
I used to do some drawings when i was younger,
but haven't been drawing since those days.
Maybe we could find someone talented out there, huh..!?

**Comment by macce2** [comment]
Sometimes there seems to be a minor screen refresh problem 
with the appearing messages (LP:s ; "X has detected the spell" etc)
when hitting enemies.
So, there maybe little too long a delay between those messages,
relatively to the recent actions on the screen.
I'm not sure if this does only occur on "faster" machines (!?);
but I have noticed this problem earlier;
especially when fighting many ("complex to draw") enemies.

**Comment by macce2** [comment]
[Feature request]  Limit the usage of potions etc. during battles to a closer range

This one might not work without doing more comprehensive modifications
to the LPs of enemies, weapons, spells etc. but I decided to write this
because this feature might make the battles somewhat more realistic
and the idea is generally taken from the game "Perihelion" and if I
remember right, it was also, to a certain degree, present in Ultima 6.

Ok in short, the idea is this:

You could only use potion(-s) and everything like that
(herbs etc) to a party member that is located physically 
close to the user.

**Comment by macce2** [comment]
[Feature request] Make enemy bosses appear as different coloured etc 
on battle map screen

There are some types of enemies that have  "chiefs" among them
(not to be mistaken for the likes of Luminor etc);
IMHO it would nice if these chiefs would appear uniquely,
compared to the "common ones" on the battle map screen;
maybe different coloured or anything like that.

**Comment by macce2** [comment]
[Request] Please move this whole thread (issue) to the Ambermoon Advanced section !?

**Comment by macce2** [comment]
Excuse me, but IMHO I must repeat this thread should better be moved to the AA
Issues section.
However, that's up to Robert, eventually.

I must tell you that I began working on some new additional
battle Backround pictures; hopefully I will have something
(very alpha  lol !) to show in the near future.
Hopefully I will get something finished and that you all
will think them to be good enough to maybe
even be used in the upcoming versions 
of Ambermoon Advanced !

**Comment by Pyrdacor** [comment]
Actually it's mostly related to the original as well. So I would leave it here. I don't have much time lately for adjustments and bugfixes but I will come back to it at a later point for sure.

**Comment by macce2** [comment]
Please consider completely re-making several 
special 2D sections with battle elements
(Bandit's House; the Madman's cellar in Spannenberg)
in 3D (Ambermoon Advanced, Ambermoon.net).
